### ISSUE TITLE
Code block inside control flow not generating properly

### ISSUE THREAD

**Issue by rgbrizzlehizzle** [enjoydambience/kotlinbard#2]
Hey, first wanna say this project is awesome, much more readable than vanilla KotlinPoet!!  

I think, however, there is currently a problem with nested control flows.  A simple example 

```
fun main() {
  val potatoFun = buildFunction("test") {
    addCode {
      controlFlow("if (1 == 1)") {
        controlFlow("if (2 == 2)") {
          addStatement("val nice = true")
          addStatement("val cool = true")
          addCode {
            addStatement("val amazing = false")
          }
        }
      }
    }
  }

  println(potatoFun.toString())
}
```

should, in theory, correspond to 

```
fun test() {
  if (1 == 1) {
    if (2 == 2) {
      val nice = true
      val cool = true
      val amazing = false
    }
  }
}
```

However, currently, KotlinBard will generate the following 

```
fun test() {
  val amazing = false
  if (1 == 1) {
    if (2 == 2) {
      val nice = true
      val cool = true
    }
  }
}
```

My project versions, if that helps, are 
kotlin -> 1.4
kotlinPoet -> 1.6.0
kotlinBard -> 0.2.0

**Comment by rgbrizzlehizzle** [comment]
ahhhh classic, just need to write up an issue to figure it out.. I need to use `add` not `addCode`.  `add` appears to remain inside the current code block context, while `addCode` remains at the function builder context, and will cause it to be generated there instead.  Curious to me that it seems to even get generated ahead of the declaring block, but I'm sure that just has to do with how KotlinPoet does code gen under the hood.  Open and close case 😄

**Comment by enjoydambience** [comment]
For documentation:

Usually this kind of accident is avoided with DslMarker annotations. There should be a `@CodegenDsl` dsl marker on all extension function scopes, but it somehow had `@Retention(SOURCE)`. This means the annotation works in tests, but wouldn't be visible if you used KotlinBard as a dependency.

Fixed in 0.3.0.
